core/templates/pages/exploration-editor-page/services/exploration-save.service.ts

```diff
@@ -158,6 +161,7 @@ angular.module('oppia').factory('ExplorationSaveService', [
             showCongratulatorySharingModal();
             SiteAnalyticsService.registerPublishExplorationEvent(
               ExplorationDataService.explorationId);
+            UserExplorationPermissionsService.getPermissionsAsync(true);
```

Comment:
exploration save service should not handle or trigger UserExplorationPermissionsService, please handle it in the directive/controller layer!

Reply:
@DubeySandeep Only these places will we know if the exploration is published or unpublished. So as soon as the exploration is published or unpublished I refresh the cache for UserExplorationPermissions.

Reply:
No, I think the calling function param `onSaveDoneCallback` can handle this, right? Can you show me where is this service's function used in the directive/component?

Reply:
So the flow is:
https://github.com/oppia/oppia/blob/a3f7191fe7a9db9ec32bc2bf1de044a8943fef57/core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.html#L11
calls:
https://github.com/oppia/oppia/blob/edfa795d011c6ce6f74ff2fe286c62617e953dad/core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.ts#L102
Inside the func `ExplorationSaveService.showPublishExplorationModal` is called which is present in:
https://github.com/oppia/oppia/blob/edfa795d011c6ce6f74ff2fe286c62617e953dad/core/templates/pages/exploration-editor-page/services/exploration-save.service.ts#L284
Which calls the following:
https://github.com/oppia/oppia/blob/edfa795d011c6ce6f74ff2fe286c62617e953dad/core/templates/pages/exploration-editor-page/services/exploration-save.service.ts#L136
inside the `showPublishExplorationModal` function

Reply:
Thanks @gp201! In that case, we can just pass the "onSaveDoneCallback" parameter as @DubeySandeep suggested here https://github.com/oppia/oppia/blob/edfa795d011c6ce6f74ff2fe286c62617e953dad/core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.ts#L106-L107.

So you'll need to pass a callback fn that will hide the loading dots as well as the permissions operation (`getPermissionsAsync`). Does this make sense?

Reply:
Thanks for the pointers @gp201, can you please follow the idea provided by @kevintab95 ? do let me know if you need any help?

Reply:
DONE